perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ExpirationNotifScheduler.java

```diff
@@ -638,11 +641,20 @@ private void expireMembers(LocalDate date) throws WrongAttributeAssignmentExcept
 			if (member.getStatus().equals(Status.VALID)) {
 				try {
 					perun.getMembersManagerBl().expireMember(sess, member);
+					List<Group> groups = sess.getPerun().getGroupsManager().getMemberGroups(sess, member);
+					for (Group g : groups) {
+						List<Application> apps = registrarManager.getApplicationsForMember(sess, g, member);
+						for (Application app: apps) {
+							registrarManager.rejectApplication(sess, app.getId(), "Application is set to status REJECTED, because the member it is assigned to is in EXPIRED state.");
```

Comment:
The problem with that is there is no easy way to call the `rejectApplication` method from the registrarManager because of the circular dependencies. I am not sure if the module rejection is also needed. I'll find out